addons/mail/tests/test_mail_gateway.py

```diff
@@ -366,6 +507,13 @@ def test_message_parse(self):
         self.assertEqual(res['body'], '')
         self.assertEqual(res['attachments'][0][0], 'thetruth.pdf')
 
+    @mute_logger('odoo.addons.mail.models.mail_thread')
+    def test_message_parse_eml(self):
+        """ Test that the parsing of mail with embedded emails as eml(msg) which generates empty attachments, can be processed.
+        """
+        with self.assertRaises(ValidationError("Processing mail with an eml attachment generates a traceback")):
```

Comment:
`with self.assertRaises` means that you expect the following code to raise an error, while your commit prevents it. Proof is that the build fails: http://runbot.odoo.com/runbot/build/326647

You could remove this line and simply try to process the email.